homeassistant/components/iometer/sensor.py

```diff
@@ -0,0 +1,160 @@
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    PERCENTAGE,
+    SIGNAL_STRENGTH_DECIBELS,
+    STATE_UNKNOWN,
+    EntityCategory,
+    UnitOfEnergy,
+    UnitOfPower,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import StateType
+
+from .coordinator import IOMeterCoordinator, IOmeterData
+from .entity import IOmeterEntity
+
+
+@dataclass(frozen=True, kw_only=True)
+class IOmeterEntityDescription(SensorEntityDescription):
+    """Describes IOmeter sensor entity."""
+
+    value_fn: Callable[[IOmeterData], str | int | float]
+
+
+SENSOR_TYPES: list[IOmeterEntityDescription] = [
+    IOmeterEntityDescription(
+        key="meter_number",
+        translation_key="meter_number",
+        icon="mdi:meter-electric",
+        value_fn=lambda data: data.status.meter.number,
+    ),
+    IOmeterEntityDescription(
+        key="connection_status",
+        translation_key="connection_status",
+        device_class=SensorDeviceClass.ENUM,
+        options=["connected", "disconnected"],
+        value_fn=lambda data: data.status.device.core.connection_status,
+    ),
+    IOmeterEntityDescription(
+        key="wifi_rssi",
+        translation_key="wifi_rssi",
+        native_unit_of_measurement=SIGNAL_STRENGTH_DECIBELS,
+        device_class=SensorDeviceClass.SIGNAL_STRENGTH,
+        state_class=SensorStateClass.MEASUREMENT,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        value_fn=lambda data: data.status.device.bridge.rssi,
+    ),
+    IOmeterEntityDescription(
+        key="core_bridge_rssi",
+        translation_key="core_bridge_rssi",
+        native_unit_of_measurement=SIGNAL_STRENGTH_DECIBELS,
+        device_class=SensorDeviceClass.SIGNAL_STRENGTH,
+        state_class=SensorStateClass.MEASUREMENT,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        value_fn=lambda data: data.status.device.core.rssi,
+    ),
+    IOmeterEntityDescription(
+        key="power_status",
+        translation_key="power_status",
+        device_class=SensorDeviceClass.ENUM,
+        options=["battery", "wired"],
+        value_fn=lambda data: data.status.device.core.power_status or STATE_UNKNOWN,
+    ),
+    IOmeterEntityDescription(
+        key="battery_level",
+        translation_key="battery_level",
+        native_unit_of_measurement=PERCENTAGE,
+        device_class=SensorDeviceClass.BATTERY,
+        state_class=SensorStateClass.MEASUREMENT,
+        value_fn=lambda data: data.status.device.core.battery_level,
+    ),
+    IOmeterEntityDescription(
+        key="attachment_status",
+        translation_key="attachment_status",
+        device_class=SensorDeviceClass.ENUM,
+        options=["attached", "detached"],
```

Comment:
should also be a binary_sensor

Reply:
This shouldn't be a binary sensor as I missed to add "unknown" to the options

Reply:
Although I just saw that the binary sensor also has the unknown state. How would we set `unknown` as a state for the binary sensor when the return value is `None` of `data.status.device.core.attachment_status`.

Reply:
you can set the state of the binary sensor to `None` and it will be shown as `Unknown`. 
alternatively you can also set `_attr_available` to `False` and it is shown as unavailable (greyed out).

Reply:
Okay, removed for now